xarray_beam/_src/pangeo_test.py

```diff
@@ -0,0 +1,219 @@
+    longitude_dim = ConcatDim('longitude', list(range(0, 144, longitude_step)))
+
+    try:
+      with tempfile.TemporaryDirectory() as tmpdir:
+        def make_path(time: int, longitude: int) -> str:
+          return f'{tmpdir}/era5-{time}-{longitude}.nc'
+
+        for time in time_dim.keys:
+          for long in longitude_dim.keys:
+            chunk = test_data.isel(
+              time=slice(time, time + time_step),
+              longitude=slice(long, long + longitude_step)
+            )
+            chunk.to_netcdf(make_path(time, long))
+        yield FilePattern(make_path, time_dim, longitude_dim)
+    finally:
+      pass
+
+  def test_prechunk_converts_correctly(self):
+    pattern = FilePattern(
+      lambda time: f"gs://bucket/{time:02d}/{time:02d}.nc",
+      ConcatDim("time", list(range(1, 31))),
+    )
+
+    transform = FilePatternToChunks(pattern)
+
+    expected = [core.Key({"time": i}) for i in range(0, 30)]
+    actual = [key for key, _ in transform._prechunk()]
+
+    self.assertEqual(expected, actual)
+
+  def test_prechunk_with_two_dims_converts_correctly(self):
+    pattern = FilePattern(
+      lambda time, level: f"gs://bucket/{time:02d}/{level:02d}.nc",
+      ConcatDim("time", list(range(1, 31))),
+      ConcatDim("level", list(range(5))),
+    )
+
+    transform = FilePatternToChunks(pattern)
+
+    expected = [core.Key({"time": i, "level": j})
+                for i, j in itertools.product(range(30), range(5))]
+    actual = [key for key, _ in transform._prechunk()]
+
+    self.assertEqual(expected, actual)
+
+  def test_prechunk_from_pattern_with_nitemsper_converts_correctly(self):
+    pattern = FilePattern(
+      lambda time, level: f"gs://bucket/{time:02d}/{level:02d}.nc",
+      ConcatDim("time", list(range(1, 31)), nitems_per_file=24),
+      ConcatDim("level", list(range(5))),
+    )
+
+    transform = FilePatternToChunks(pattern)
+
+    expected = [core.Key({"time": i, "level": j})
+                for i, j in itertools.product(range(0, 30 * 24, 24), range(5))]
+    actual = [key for key, _ in transform._prechunk()]
+
+    self.assertEqual(expected, actual)
+
+  def test_no_subchunks_returns_single_dataset(self):
+    expected = [(core.Key(_zero_dimensions(self.TEST_DATA)), self.TEST_DATA)]
+    with self.pattern_from_testdata() as pattern:
+      actual = test_util.EagerPipeline() | FilePatternToChunks(pattern)
+
+    self.assertIdenticalChunks(actual, expected)
+
+  def test_single_subchunks_returns_multiple_datasets(self):
+    base_key = core.Key(_zero_dimensions(self.TEST_DATA))
+
+    with self.pattern_from_testdata() as pattern:
+      result = (
+          test_util.EagerPipeline()
+          | FilePatternToChunks(pattern, sub_chunks={"longitude": 48})
+      )
+
+    expected_keys = [base_key.with_offsets(longitude=i)
+                     for i in range(0, 144, 48)]
+    expected_sizes = [{"time": 365 * 4, "latitude": 73, "longitude": 48}
+                      for _ in range(3)]
```

Comment:
Rather than only checking sizes, why not explicitly construct expected outputs and use `assertIdenticalChunks`? I suspect that would not be much harder with `isel`?